Data Race and nil pointer on systray `connect` button

**Describe the bug**
Nil pointer and data race in systray **Connect** button.

**Environment information:**
- OS: e.g. Mac OS X

**Steps to Reproduce**
Steps to reproduce the behavior:

1. Build systray with `-race` on skywire-visor
2. Run a visor
3. Click **Connect** on systray menu
4. See errors

**Actual behavior**
```
==================
WARNING: DATA RACE
Write at 0x0001066271a8 by goroutine 21:
  github.com/skycoin/skywire/internal/gui.handleVPNButton()
      /Users/mohammed/Projects/Skycoin/skywire/internal/gui/gui.go:271 +0x94
  github.com/skycoin/skywire/internal/gui.handleUserInteraction()
      /Users/mohammed/Projects/Skycoin/skywire/internal/gui/gui.go:390 +0x258

Previous read at 0x0001066271a8 by goroutine 126:
  github.com/skycoin/skywire/internal/gui.vpnStatusBtn()
      /Users/mohammed/Projects/Skycoin/skywire/internal/gui/gui.go:217 +0x318

Goroutine 21 (running) created at:
  github.com/skycoin/skywire/internal/gui.GetOnGUIReady.func2()
      /Users/mohammed/Projects/Skycoin/skywire/internal/gui/gui.go:92 +0x15c
  github.com/skycoin/systray.Register.func2()
      /Users/mohammed/Projects/Skycoin/skywire/vendor/github.com/skycoin/systray/systray.go:99 +0x40

Goroutine 126 (running) created at:
  github.com/skycoin/skywire/internal/gui.initVpnClientBtn()
      /Users/mohammed/Projects/Skycoin/skywire/internal/gui/gui.go:187 +0x2fc
  github.com/skycoin/skywire/internal/gui.GetOnGUIReady.func2()
      /Users/mohammed/Projects/Skycoin/skywire/internal/gui/gui.go:89 +0xbc
  github.com/skycoin/systray.Register.func2()
      /Users/mohammed/Projects/Skycoin/skywire/vendor/github.com/skycoin/systray/systray.go:99 +0x40
==================
[2022-05-16T14:16:16+04:30] INFO [visor_rpc:CLI]: Request processed. _elapsed="92.042µs" _method="StartApp" _received="2:16PM" input=0xc00031e1e0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x104f1471c]

goroutine 318 [running]:
github.com/skycoin/skywire/pkg/transport.(*Manager).STCPRRemoteAddrs(0x0)
	/Users/mohammed/Projects/Skycoin/skywire/pkg/transport/manager.go:459 +0x3c
github.com/skycoin/skywire/pkg/visor.(*Visor).StartApp(0xc0002a8000, {0xc000323b86, 0xa})
	/Users/mohammed/Projects/Skycoin/skywire/pkg/visor/api.go:318 +0x98
github.com/skycoin/skywire/pkg/visor.(*RPC).StartApp(0xc000530040, 0xc00031e1e0, 0x1066273e0)
	/Users/mohammed/Projects/Skycoin/skywire/pkg/visor/rpc.go:213 +0x120
reflect.Value.call({0xc00051a8a0, 0xc0005363f8, 0x13}, {0x10500f0e2, 0x4}, {0xc00014af10, 0x3, 0x3})
	/usr/local/go/src/reflect/value.go:556 +0x868
reflect.Value.Call({0xc00051a8a0, 0xc0005363f8, 0x13}, {0xc0002c0710, 0x3, 0x3})
	/usr/local/go/src/reflect/value.go:339 +0x9c
net/rpc.(*service).call(0xc0005320c0, 0xc0005340a0, 0xc00058c2e8, 0xc00058c2f4, 0xc000539080, 0xc0002961e0, {0x10558e440, 0xc00031e1e0, 0x16}, {0x10558e900, ...}, ...)
	/usr/local/go/src/net/rpc/server.go:377 +0x198
created by net/rpc.(*Server).ServeCodec
	/usr/local/go/src/net/rpc/server.go:474 +0x34c
```

--- a/internal/gui/gui.go
+++ b/internal/gui/gui.go
@@ -61,6 +61,7 @@ var (
 	mVPNButton      *systray.MenuItem
 	mUninstall      *systray.MenuItem
 	mQuit           *systray.MenuItem
+	vpnStatusMx     sync.Mutex
 )
 
 // GetOnGUIReady creates func to run on GUI startup.
@@ -198,6 +199,7 @@ func initVpnClientBtn(conf *visorconfig.V1, httpClient *http.Client, logger *log
 
 func vpnStatusBtn(conf *visorconfig.V1, rpcClient visor.API) {
 	for {
+		vpnStatusMx.Lock()
 		stats, _ := rpcClient.GetAppConnectionsSummary(skyenv.VPNClientName)
 		if len(stats) == 1 {
 			if stats[0].IsAlive {
@@ -224,6 +226,7 @@ func vpnStatusBtn(conf *visorconfig.V1, rpcClient visor.API) {
 				vpnLastStatus = 0
 			}
 		}
+		vpnStatusMx.Unlock()
 		time.Sleep(2 * time.Second)
 	}
 }
@@ -258,7 +261,9 @@ func serversBtn(conf *visorconfig.V1, servers []*systray.MenuItem, rpcClient vis
 
 		rpcClient.StopApp(skyenv.VPNClientName)
 		rpcClient.SetAppPK(skyenv.VPNClientName, pk)
+		vpnStatusMx.Lock()
 		vpnLastStatus = 3
+		vpnStatusMx.Unlock()
 		rpcClient.StartApp(skyenv.VPNClientName)
 	}
 }
@@ -268,7 +273,9 @@ func handleVPNButton(conf *visorconfig.V1, rpcClient visor.API) {
 	if len(stats) == 1 {
 		rpcClient.StopApp(skyenv.VPNClientName)
 	} else {
+		vpnStatusMx.Lock()
 		vpnLastStatus = 3
+		vpnStatusMx.Unlock()
 		rpcClient.StartApp(skyenv.VPNClientName)
 	}
 }

--- a/pkg/visor/api.go
+++ b/pkg/visor/api.go
@@ -315,6 +315,11 @@ func (v *Visor) StartApp(appName string) error {
 	if appName == skyenv.VPNClientName {
 		// todo: can we use some kind of app start hook that will be used for both autostart
 		// and start? Reason: this is also called in init for autostart
+
+		// check transport manager availability
+		if v.tpM == nil {
+			return ErrTrpMangerNotAvailable
+		}
 		maker := vpnEnvMaker(v.conf, v.dmsgC, v.dmsgDC, v.tpM.STCPRRemoteAddrs())
 		envs, err = maker()
 		if err != nil {
@@ -325,14 +330,21 @@ func (v *Visor) StartApp(appName string) error {
 			return errors.New("VPN server pub key is missing")
 		}
 	}
-
-	return v.appL.StartApp(appName, nil, envs)
+	// check process manager availability
+	if v.procM != nil {
+		return v.appL.StartApp(appName, nil, envs)
+	}
+	return ErrProcNotAvailable
 }
 
 // StopApp implements API.
 func (v *Visor) StopApp(appName string) error {
-	_, err := v.appL.StopApp(appName) //nolint:errcheck
-	return err
+	// check process manager availability
+	if v.procM != nil {
+		_, err := v.appL.StopApp(appName) //nolint:errcheck
+		return err
+	}
+	return ErrProcNotAvailable
 }
 
 // SetAppDetailedStatus implements API.
@@ -540,6 +552,7 @@ func (v *Visor) GetAppError(appName string) (string, error) {
 
 // GetAppConnectionsSummary implements API.
 func (v *Visor) GetAppConnectionsSummary(appName string) ([]appserver.ConnectionSummary, error) {
+	// check process manager availability
 	if v.procM != nil {
 		cSummary, err := v.procM.ConnectionsSummary(appName)
 		if err != nil {
@@ -548,7 +561,7 @@ func (v *Visor) GetAppConnectionsSummary(appName string) ([]appserver.Connection
 
 		return cSummary, nil
 	}
-	return nil, nil
+	return nil, ErrProcNotAvailable
 }
 
 // TransportTypes implements API.

--- a/pkg/visor/visor.go
+++ b/pkg/visor/visor.go
@@ -34,6 +34,10 @@ import (
 var (
 	// ErrAppProcNotRunning represents lookup error for App related calls.
 	ErrAppProcNotRunning = errors.New("no process of given app is running")
+	// ErrProcNotAvailable represents error for unavailable process manager
+	ErrProcNotAvailable = errors.New("no process manager not available")
+	// ErrTrpMangerNotAvailable represents error for unavailable transport manager
+	ErrTrpMangerNotAvailable = errors.New("no transport manager not available")
 )
 
 const (

--- a/pkg/visor/visor.go
+++ b/pkg/visor/visor.go
@@ -35,9 +35,9 @@ var (
 	// ErrAppProcNotRunning represents lookup error for App related calls.
 	ErrAppProcNotRunning = errors.New("no process of given app is running")
 	// ErrProcNotAvailable represents error for unavailable process manager
-	ErrProcNotAvailable = errors.New("no process manager not available")
+	ErrProcNotAvailable = errors.New("no process manager available")
 	// ErrTrpMangerNotAvailable represents error for unavailable transport manager
-	ErrTrpMangerNotAvailable = errors.New("no transport manager not available")
+	ErrTrpMangerNotAvailable = errors.New("no transport manager available")
 )
 
 const (

